### DISCUSSION TITLE
Does ginkgo support that running on AMD GPU with dpcpp backend?

### DISCUSSION THREAD

**Discussion by AvailableCXGuo** [ginkgo-project/ginkgo#1623] [Q&A]
Hi there, I am trying to run ginkgo on AMD GPU with dpcpp, since Intel dpcpp supports HIP now. But I got segment fault when I ran the example "simple solver" and fixed the exec in the .cpp with "const auto exec = gko::DpcppExecutor::create(0, gko::OmpExecutor::create(), "gpu");". I know that the dpcpp one can be run on CPUs. How about the GPUs?

**Comment by MarcelKoch** [comment]
I'm not aware of anyone who has tried this before, so you are on novel ground here. Could you perhaps provide a stack trace for the segfault?
Also, did you try to run a sycl program (without any ginkgo in it) on you AMD GPU?

**Reply by AvailableCXGuo** [reply]
I tried the example of simple_solver that Ginkgo provides with the command ./simple_solver dpcpp and it was good. But I realized the example was running on the intel cpu. So I modified the code of simple_solver.cpp "return gko::DpcppExecutor::create(0,gko::OmpExecutor::create());" at line 73 with  "return gko::DpcppExecutor::create(0,gko::OmpExecutor::create(), "gpu");". Then I got the error of INVALID_BINARY. My AMD GPU is MI210.

**Reply by AvailableCXGuo** [reply]
I tried the simple_sycl.cpp that comes from intel llvm documentation. It runs well on AMD GPU.

**Reply by MarcelKoch** [reply]
Could you try running
```
std::cout << "GPUs detected by SYCL: " << gko::DpcppExecutor::get_num_devices("gpu") << std::endl;
```
before you create the executor?
